src/hooks/dapps-staking/useCompoundRewards.ts

```diff
@@ -76,9 +78,11 @@ export function useCompoundRewards() {
 
   const setRewardDestination = async (rewardDestination: RewardDestination): Promise<void> => {
     try {
-      const dappStakingService = container.get<IDappStakingService>(Symbols.DappStakingService);
+      const dappStakingService = container.get<IDappStakingService>(
```

Comment:
Instead having this code on multiple you can do something similar we did for `IWalletService`
```
container.bind<interfaces.Factory<IWalletService>>(Symbols.WalletFactory).toFactory(() => {
    return () => {
      return container.get<IWalletService>(currentWalletType);
    };
  });
```
After that you can use factory to get proper instance of service

Reply:
Thanks, I fixed it in [this](https://github.com/AstarNetwork/astar-apps/pull/857/commits/53e2951cf9c6eb8f63b673304086e82c493479ce) commit.